### DISCUSSION TITLE
Getting an LSP working in dune

### DISCUSSION THREAD

**Discussion by H-ANSEN** [ocaml/dune#10756] [Q&A] [answered]
I'm using Neovim with `ocamllsp` and `lsp-zero`. I'm having some difficulties getting my LSP to run while in the Dune code base. I was wondering if anyone else has a similar setup and has resolved this issue in the past? I am notably getting the following diagnostic message at the top of each source file:

> The current Merlin configuration has been generated by another, incompatible, version of Dune. Please rebuild the project. *(Using the same version of Dune as the one running the `ocaml-merlin` server.)*

I assume that this has to do with the whole *Dune building Dune* aspect, but sense I am running the LSP through `lsp-zero` I'm unsure how I would go about handling Merlin.

**Comment by rgrinberg** [comment, marked as answer]
The correct steps should be:

```
$ make dev
$ ./dune.exe build @install
$ ./dune.exe exec -- nvim
```

The launched neovim should have the correct dune in its environment.

**Reply by H-ANSEN** [reply]
Thank you! Works perfectly 😸
